### PR TITLE
[Media] Adding additional customization files

### PR DESCRIPTION
**Pull request by hivyas**
Updated public swagger link and added MediaGraphNodeInput.cs and MediaGraphPemCertificateList.cs

### CODE REVIEW

File: sdk/mediaservices/Azure.Media.Analytics.Edge/src/Customization/MediaGraphInstanceSetRequest.cs [hunk 1]
```diff
@@ -5,20 +5,18 @@ namespace Azure.Media.Analytics.Edge.Models
 {
     public partial class MediaGraphInstanceSetRequest
     {
-        internal MediaGraphInstanceSetRequestBody InstanceBody;
+        private MediaGraphInstanceSetRequestBody _instanceBody;
```
Comment (pakrym) [inline sdk/mediaservices/Azure.Media.Analytics.Edge/src/Customization/MediaGraphInstanceSetRequest.cs:8]: Do you need this field? Can it be a local?

File: sdk/mediaservices/Azure.Media.Analytics.Edge/src/Customization/MediaGraphInstanceCollection.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    public partial class MediaGraphInstanceCollection
+    {
+        /// <summary>
+        ///  Deserialize MediaGraphInstance.
+        /// </summary>
+        /// <param name="json">The json to be deserialized.</param>
+        /// <returns> Deserialized Graph Instance Collection. </returns>
+        public static MediaGraphInstanceCollection Deserialize(string json)
+        {
+            JsonElement element = JsonDocument.Parse(json).RootElement;
```
Comment (pakrym) [inline sdk/mediaservices/Azure.Media.Analytics.Edge/src/Customization/MediaGraphInstanceCollection.cs:20]: JsonDocument is disposable and needs to be disposed.

File: sdk/mediaservices/Azure.Media.Analytics.Edge/api/Azure.Media.Analytics.Edge.netstandard2.0.cs [hunk 3]
```diff
@@ -259,21 +273,53 @@ public partial class MediaGraphMotionDetectionProcessor : Azure.Media.Analytics.
     }
     public partial class MediaGraphNodeInput
     {
-        public MediaGraphNodeInput() { }
+        public MediaGraphNodeInput(string nodeName) { }
         public string NodeName { get { throw null; } set { } }
```
Comment (pakrym) [inline sdk/mediaservices/Azure.Media.Analytics.Edge/api/Azure.Media.Analytics.Edge.netstandard2.0.cs:277]: If you are taking the `NodeName` as a ctor parameter the property shouldn't be settable.

File: sdk/mediaservices/Azure.Media.Analytics.Edge/src/Customization/MediaGraphPemCertificateList.cs [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+            if (certificates == null)
+            {
+                throw new ArgumentNullException(nameof(certificates));
+            }
+
+            Certificates = certificates.Select(ConvertToPemString).ToList();
+            Type = "#Microsoft.Media.MediaGraphPemCertificateList";
+        }
+
+        private static string ConvertToPemString(X509Certificate2 certificate)
```
Comment (pakrym) [inline sdk/mediaservices/Azure.Media.Analytics.Edge/src/Customization/MediaGraphPemCertificateList.cs:68]: Do we have a test for this method?

File: sdk/mediaservices/Azure.Media.Analytics.Edge/src/Customization/MediaGraphTopologySetRequest.cs [hunk 5]
```diff
@@ -7,20 +7,18 @@ namespace Azure.Media.Analytics.Edge.Models
 {
     public partial class MediaGraphTopologySetRequest
     {
-        internal MediaGraphTopologySetRequestBody GraphBody;
+        private MediaGraphTopologySetRequestBody _graphBody;
```
Comment (pakrym) [inline sdk/mediaservices/Azure.Media.Analytics.Edge/src/Customization/MediaGraphTopologySetRequest.cs:10]: looks like this field can be removed as well

**Comment by pakrym** [comment]
LGTM

File: sdk/mediaservices/Azure.Media.Analytics.Edge/src/Customization/MediaGraphPemCertificateList.cs (hunk 4 above)
Comment (hivyas) [reply sdk/mediaservices/Azure.Media.Analytics.Edge/src/Customization/MediaGraphPemCertificateList.cs:68]: The tests for this are currently in the service repo. We have plans to move some of them to this repo in the future.

File: sdk/mediaservices/Azure.Media.Analytics.Edge/api/Azure.Media.Analytics.Edge.netstandard2.0.cs (hunk 3 above)
Comment (hivyas) [reply sdk/mediaservices/Azure.Media.Analytics.Edge/api/Azure.Media.Analytics.Edge.netstandard2.0.cs:277]: This is autogenerated code. Not sure that we can control this unless we change the actual autorest file itself?

File: sdk/mediaservices/Azure.Media.Analytics.Edge/src/Customization/MediaGraphInstanceSetRequest.cs (hunk 1 above)
Comment (hivyas) [reply sdk/mediaservices/Azure.Media.Analytics.Edge/src/Customization/MediaGraphInstanceSetRequest.cs:8]: Fixed

File: sdk/mediaservices/Azure.Media.Analytics.Edge/src/Customization/MediaGraphTopologySetRequest.cs (hunk 5 above)
Comment (hivyas) [reply sdk/mediaservices/Azure.Media.Analytics.Edge/src/Customization/MediaGraphTopologySetRequest.cs:10]: Fixed

File: sdk/mediaservices/Azure.Media.Analytics.Edge/api/Azure.Media.Analytics.Edge.netstandard2.0.cs (hunk 3 above)
Comment (pakrym) [reply sdk/mediaservices/Azure.Media.Analytics.Edge/api/Azure.Media.Analytics.Edge.netstandard2.0.cs:277]: if you copy the property into your customization file the generator wouldn't generate it anymore and you can customize it at will.

Comment (hivyas) [reply sdk/mediaservices/Azure.Media.Analytics.Edge/api/Azure.Media.Analytics.Edge.netstandard2.0.cs:277]: This is happening in many places though. For example look at the property instance in MediaGraphInstanceSetRequest.cs.

Comment (pakrym) [reply sdk/mediaservices/Azure.Media.Analytics.Edge/api/Azure.Media.Analytics.Edge.netstandard2.0.cs:277]: Ah, my bad. I was under the impression you've added a customized ctor. Ignore me.

**Review by giakas — APPROVED** [review]
LGTM

**Pull request merged**